src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -10,31 +10,39 @@ import html from '../../internals/rich-text-utils/html';
 import isEmpty from '../../internals/rich-text-utils/is-empty';
 
 class RichTextInput extends React.PureComponent {
-  state = {
-    isFocused: false,
-  };
-
-  onValueChange = ({ value }) => {
-    const event = {
+  onValueChange = event => {
+    const fakeEvent = {
       target: {
         id: this.props.id,
         name: this.props.name,
-        value,
+        value: event.value,
       },
     };
-    this.props.onChange(event);
+
+    this.props.onChange(fakeEvent);
   };
 
   // this issue explains why we need to use next() + setTimeout
   // for calling our passed onBlur handler
   // https://github.com/ianstormtaylor/slate/issues/2434
   onBlur = (event, editor, next) => {
+    // we don't call next() if it's a button to stop our input from losing
+    // slate focus
+    if (
+      event.relatedTarget &&
+      event.relatedTarget.getAttribute('data-button-type') ===
```

Comment:
we prevent slate from firing `onBlur` when it is triggered by a button within the component's toolbar.